### ISSUE TITLE
Response includes "Transfer-Encoding: chunked" and "Content-Length"

### ISSUE THREAD

**Issue by gajus** [mitmproxy/mitmproxy#1033]
##### Steps to reproduce the problem:

1. Make request to get http://m.cinemasgaumontpathe.com/cinemas/.

```
GET /cinemas/ HTTP/1.1
encoding: iso-8859-1
host: m.cinemasgaumontpathe.com
Connection: close

```

##### What is the expected behavior?

Expected to see original headers, without `Transfer-Encoding: chunked`, e.g.

```sh
$ curl -I http://m.cinemasgaumontpathe.com/cinemas/
HTTP/1.1 200 OK
Server: Apache
Pragma: no-cache
Last-Modified: Wed, 16 Mar 2016 12:46:15 +0100
Content-Type: text/html; charset=iso-8859-1
Cache-Control: no-cache, no-store
Date: Wed, 16 Mar 2016 11:46:15 GMT
Connection: keep-alive
```

##### What went wrong?

Response includes `Transfer-Encoding: chunked` and `Content-Length` headers.

```
HTTP/1.1 200 OK
Server: Apache
Pragma: no-cache
Last-Modified: Wed, 16 Mar 2016 12:46:55 +0100
Content-Type: text/html; charset=iso-8859-1
X-Akamai-Transformed: 9 5509 0 pmb=mRUM,1
Cache-Control: no-cache, no-store
Date: Wed, 16 Mar 2016 11:46:55 GMT
Transfer-Encoding: chunked
Connection: close
Connection: Transfer-Encoding
Vary: Accept-Encoding, User-Agent
content-length: 33907

8473
```
##### Any other comments?

* Chunked transfer encoding must not include `Content-Length` header (https://en.wikipedia.org/wiki/Chunked_transfer_encoding).
* Related issue https://github.com/request/request/issues/2091

---

Mitmproxy Version: v0.15
Operating System: OSX v10.11.3

**Comment by mhils** [comment]
Thanks for the report. Could you please update to mitmproxy 0.16 and see if that still happens? At least on master, I get this:
```
λ curl -I http://m.cinemasgaumontpathe.com/cinemas/ -x localhost:8080
HTTP/1.1 200 OK
Server: Apache
Pragma: no-cache
Last-Modified: Wed, 16 Mar 2016 12:58:24 +0100
Content-Type: text/html; charset=iso-8859-1
Cache-Control: no-cache, no-store
Date: Wed, 16 Mar 2016 11:58:24 GMT
Connection: close

~
λ curl -I http://m.cinemasgaumontpathe.com/cinemas/
HTTP/1.1 200 OK
Server: Apache
Pragma: no-cache
Last-Modified: Wed, 16 Mar 2016 12:58:33 +0100
Content-Type: text/html; charset=iso-8859-1
Cache-Control: no-cache, no-store
Date: Wed, 16 Mar 2016 11:58:33 GMT
Connection: keep-alive
```

**Comment by gajus** [comment]
Is there a pre-built binary for v0.16?

**Comment by mhils** [comment]
Whoops. Looks like we missed updating mitmproxy.org. :open_mouth:
Binaries are at https://snapshots.mitmproxy.org/!

Cheers,
Max

**Comment by gajus** [comment]
When you are doing `curl -I` you are making HEAD request. Try `curl -i ` instead.

```sh
$ curl -i http://m.cinemasgaumontpathe.com/cinemas/ -x 127.0.0.1:8082
HTTP/1.1 200 OK
Server: Apache
Pragma: no-cache
Last-Modified: Wed, 16 Mar 2016 13:07:06 +0100
Content-Type: text/html; charset=iso-8859-1
Cache-Control: no-cache, no-store
Date: Wed, 16 Mar 2016 12:07:06 GMT
Transfer-Encoding: chunked
Connection: close
Connection: Transfer-Encoding
Vary: Accept-Encoding, User-Agent
content-length: 32867
```

```sh
$ curl -i http://m.cinemasgaumontpathe.com/cinemas/
HTTP/1.1 200 OK
Server: Apache
Pragma: no-cache
Last-Modified: Wed, 16 Mar 2016 13:07:29 +0100
Content-Type: text/html; charset=iso-8859-1
Cache-Control: no-cache, no-store
Date: Wed, 16 Mar 2016 12:07:29 GMT
Transfer-Encoding:  chunked
Connection: keep-alive
Connection: Transfer-Encoding
Vary: Accept-Encoding, User-Agent
```

**Comment by gajus** [comment]
Using mitmproxy v0.16:

```sh
$ curl -i http://m.cinemasgaumontpathe.com/cinemas/ -x 127.0.0.1:8082
HTTP/1.1 200 OK
Server: Apache
Pragma: no-cache
Last-Modified: Wed, 16 Mar 2016 13:09:27 +0100
Content-Type: text/html; charset=iso-8859-1
Cache-Control: no-cache, no-store
Date: Wed, 16 Mar 2016 12:09:27 GMT
Transfer-Encoding: chunked
Connection: close
Connection: Transfer-Encoding
Vary: Accept-Encoding, User-Agent
```

**Comment by gajus** [comment]
Appear to be fixed in v0.16.

Thank you.
